lib/smart_answer_flows/maternity-paternity-calculator/maternity_calculator_flow.rb

```diff
@@ -124,24 +105,17 @@ def define
           from { 2.years.ago(Time.zone.today) }
           to { 2.years.since(Time.zone.today) }
 
-          precalculate :payday_offset do
-            calculator.payday_offset
-          end
-
-          precalculate :payday_offset_formatted do
-            calculator.format_date_day payday_offset
-          end
-
-          calculate :last_payday_eight_weeks do |response|
-            payday = response + 1.day
-            raise SmartAnswer::InvalidResponse if payday > payday_offset
+          on_response do |response|
+            self.last_payday_eight_weeks = 1.day.after(response)
+            calculator.pre_offset_payday = last_payday_eight_weeks
 
-            calculator.pre_offset_payday = payday
-            payday
+            self.payday_offset = calculator.payday_offset
+            self.payday_offset_formatted = calculator.payday_offset_formatted
```

Comment:
I don't think we need these assignments. Changing the validation block to use`calculator.payday_offset` should prevent the tests from breaking.

Reply:
Yes - you're right. These have been removed and the tests are passing.